arangod/Aql/Functions.cpp

```diff
@@ -8848,6 +8855,132 @@ AqlValue Functions::MakeDistributeGraphInput(arangodb::aql::ExpressionContext* e
+  AqlValue const& scale_value = extractFunctionParameterValue(parameters, 2);
+  AqlValue const& offset_value = extractFunctionParameterValue(parameters, 3);
+  AqlValue const& decay_value = extractFunctionParameterValue(parameters, 4);
+
+  // check type of arguments
+  if ((!arg_value.isRange() && !arg_value.isArray() && !arg_value.isNumber()) ||
+    !origin_value.isNumber() || !scale_value.isNumber() ||
+    !offset_value.isNumber() || !decay_value.isNumber()) {
+
+    registerInvalidArgumentWarning(expressionContext, getFunctionName(node).c_str());
+    return AqlValue(AqlValueHintNull());
+  }
+
+  // extracting values
+  bool failed;
+  bool error = false;
+  double origin = origin_value.toDouble(failed);
+  error |= failed;
+  double scale = scale_value.toDouble(failed);
+  error |= failed;
+  double offset = offset_value.toDouble(failed);
+  error |= failed;
+  double decay = decay_value.toDouble(failed);
+  error |= failed;
+
+  if (error) {
+    registerWarning(expressionContext, getFunctionName(node).c_str(), TRI_ERROR_QUERY_FUNCTION_ARGUMENT_TYPE_MISMATCH);
+    return AqlValue(AqlValueHintNull());
+  }
+
+  // check that parameters are correct
+  if (scale <= 0 || offset < 0 || decay <= 0 || decay >= 1) {
+    registerWarning(expressionContext, getFunctionName(node).c_str(), TRI_ERROR_QUERY_NUMBER_OUT_OF_RANGE);
+    return AqlValue(AqlValueHintNull());
+  }
+
+  // get lambda for further calculation
+  auto f = lambda_func(origin, scale, offset, decay);
+
+  // argument is number
+  if (arg_value.isNumber()) {
+      double arg = arg_value.slice().getNumber<double>();
+      double func_res = f(arg);
+      return ::numberValue(func_res, true);
+  } else {
+    // argument is array or range
+    auto* trx = &expressionContext->trx();
+    AqlValueMaterializer materializer(&trx->vpackOptions());
+    VPackSlice slice = materializer.slice(arg_value, true);
+    TRI_ASSERT(slice.isArray());
+
+    VPackBuilder builder;
+    {
+      VPackArrayBuilder array_builder(&builder);
+      for (VPackSlice curr_arg : VPackArrayIterator(slice)) {
+        if (!curr_arg.isNumber()) {
+          registerWarning(expressionContext, getFunctionName(node).c_str(), TRI_ERROR_QUERY_FUNCTION_ARGUMENT_TYPE_MISMATCH);
+          return AqlValue(AqlValueHintNull());
+        }
+        double arg = curr_arg.getNumber<double>();
+        double func_res = f(arg);
+        builder.add(VPackValue(func_res));
+      }
+    }
+
+    return AqlValue(builder.slice());
+  }
+}
+
+AqlValue Functions::GaussDecay(arangodb::aql::ExpressionContext* expressionContext,
+                                             AstNode const& node,
+                                             VPackFunctionParameters const& parameters) {
+
+  auto gauss_decay_f = [](const double origin, const double scale, const double offset, const double decay) {
+    double sigma_sqr = - (scale * scale) / (2 * std::log(decay));
+    return [=](double arg) {
```

Comment:
default capture is bad practice, please specify variables that are necessary for evaluation